### PR TITLE
Visual alert for Tram Malfunction event

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request
Tram Malfunction event now has a BSOD similar to the cargo signs for the duration it runs.

tram-sad
image
## Why It's Good For The Game
Shows tram crossing signal needs to be updated to Windows 11.
## Changelog
:cl: LT3
add: Tram crossing signals now visually show when they're malfunctioning.
fix: Emagged tram crossing signals will stay emagged after the tram malfunction event.
/:cl:

### CODE REVIEW

File: code/modules/events/tram_malfunction.dm [hunk 1]
```diff
@@ -42,8 +42,8 @@
 
 		signal.obj_flags |= EMAGGED
 
-		if(signal.signal_state != XING_STATE_GREEN)
-			signal.set_signal_state(XING_STATE_GREEN)
+		if(signal.signal_state != XING_STATE_MALF)
```
Comment (Mothblocks) [inline code/modules/events/tram_malfunction.dm:45]: This should be in a shared defines file instead of copying and pasting between the two

But more likely this should just be a proc like `signal.set_malfunctioning()`

**Comment by Elli-Skala** [comment]
Will this effect emagging it?

**Comment by Jacquerel** [comment]
> Will this effect emagging it?

Funnily enough I am pretty sure that the tram malfunction event un-emags all emagged signs after running.

**Comment by lessthnthree** [comment]
> > Will this effect emagging it?
> 
> Funnily enough I am pretty sure that the tram malfunction event un-emags all emagged signs after running.

It shouldn't, but I'll test this edge case now and either come back with a fix or nothing.

**Review by Twaticus — APPROVED** [review]

File: code/modules/events/tram_malfunction.dm (hunk 1 above)
Comment (lessthnthree) [reply code/modules/events/tram_malfunction.dm:45]: Done

**Comment by lessthnthree** [comment]
Fixed the emagging thing too, and reduced the collision lethality multiplier since https://github.com/tgstation/tgstation/pull/71754 is coming.

**Comment by Zergspower** [comment]
:<
